support-frontend/assets/helpers/redux/checkout/product/contributionsSideEffects.ts

```diff
@@ -75,7 +75,7 @@ export function addProductSideEffects(
 	});
 
 	startListening({
-		type: 'SET_CHECKOUT_FORM_HAS_BEEN_SUBMITTED',
+		type: 'PAYMENT_WAITING',
```

Comment:
I've completely removed the `setCheckoutFormHasBeenSubmitted` action as we no longer use that field in Redux; however it had an important side effect of setting the contribution amount in session storage for us to pick up and send to Quantum Metric following a one-off PayPal contribution (which involves a redirect via PayPal, meaning we lose all Redux state).

I've attached this side effect to the `paymentWaiting` action creator for now, as that is called before we redirect to PayPal. However it might be better to just make this an explicit effect of clicking the one-off PayPal payment button?

Reply:
> However it might be better to just make this an explicit effect of clicking the one-off PayPal payment button?

Would this be to simply run this within an `onClick` handler to the PayPal payment button?

Reply:
Yes, although on further investigation I couldn't find anywhere we were actually retrieving the `contributionAmount` key from session storage- I'm not sure if this is intentional? If so we could remove this side effect entirely.

Reply:
It's retrieved here https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouPage.tsx#L35, so we still need it, but moving the logic to a `onClick` handler seems reasonable if this is the only event which has this side effect.

Reply:
That page is no longer in use! I assume that needs to be replicated on the new thank you pages? We really do need all this Quantum Metric stuff documented somewhere.

If we're going to assume that `contributionAmount` will _always_ be in session storage- as that code does- we need to make sure to always put it there, so it _shouldn't_ be set purely in a click handler for one-off PayPal payment. We could use it as a fallback for retrieving the amount from Redux, but that would involve a bit more complexity than just copying and pasting over that old code.

Reply:
OK it looks like there's a bug on [supporterPlusThankYou.tsx]( the https://github.com/guardian/support-frontend/blob/00468e02176ea94faed4a3982c2a585befcf8527/support-frontend/assets/pages/supporter-plus-thank-you/supporterPlusThankYou.tsx) for One Off Paypal payments, where it should be pulling the `contributionAmount` from session storage, but it isn't and therefore it's reporting the wrong amount...

Reply:
When defining `amount` we should check `isOneOffPayPal` and if `isOneOffPayPal` is `true` we use the `contributionAmount` from session storage rather than the `getAmount` helper.